frontend/src/component/common/Sticky/Sticky.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+    HTMLAttributes,
+    ReactNode,
+    useContext,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
+import { StickyContext } from './StickyContext';
+import { styled } from '@mui/material';
+
+const StyledSticky = styled('div', {
+    shouldForwardProp: (prop) => prop !== 'top',
+})<{ top?: number }>(({ theme, top }) => ({
+    position: 'sticky',
+    zIndex: theme.zIndex.sticky - 100,
+    ...(top !== undefined
+        ? {
+              '&': {
+                  top,
+              },
+          }
+        : {}),
+}));
+
+interface IStickyProps extends HTMLAttributes<HTMLDivElement> {
+    children: ReactNode;
+}
+
+export const Sticky = ({ children, ...props }: IStickyProps) => {
+    const context = useContext(StickyContext);
+    const ref = useRef<HTMLDivElement>(null);
+    const [initialTopOffset, setInitialTopOffset] = useState<number | null>(
+        null,
+    );
+    const [top, setTop] = useState<number>();
+
+    if (!context) {
+        throw new Error(
+            'Sticky component must be used within a StickyProvider',
+        );
+    }
+
+    const {
+        registerStickyItem,
+        unregisterStickyItem,
+        getTopOffset,
+        stickyItems,
+    } = context;
+
+    useEffect(() => {
```

Comment:
as someone who hasn't written this code I have no clue what each of the useEffects is doing. Intention revealing names + comments would help

Reply:
Let me know if this helps: https://github.com/Unleash/unleash/pull/5088/commits/855fd941d7006d760ac7c7358a4320d2d3ad29a0